feat: add upgrade check (#1444)
denoland/fresh: docs/canary/concepts/updating.md

---
description: |

>   Fresh regularly releases new versions of the framework. This page explains how to update your project.

Reviewer: ```suggestion
  New versions of Fresh are regularly released. This page explains how to update your project.
```
Nit: it may sound like "Fresh" and "the framework" are two separate entities.

project:

```bash
$ deno run -A -r https://fresh.deno.dev/update .
```

You will be prompted to confirm the changes that will be made to your project.

### Manual update

To manually update your project's dependencies, you can edit the `deno.json`
file in the root of your projects directory. Dependency versions are encoded
into the URLs in this file. For example, here is how to update a project from
Fresh 1.0.2 to 1.1.3, and update Preact to the latest version:

```diff
  {
    "imports": {
-     "$fresh/": "https://deno.land/x/fresh@1.0.2/",
+     "$fresh/": "https://deno.land/x/fresh@1.1.5/",

-     "preact": "https://esm.sh/preact@10.8.1",
-     "preact/": "https://esm.sh/preact@10.8.1/",
+     "preact": "https://esm.sh/preact@10.11.0",
+     "preact/": "https://esm.sh/preact@10.11.0/",

-     "preact-render-to-string": "https://esm.sh/*preact-render-to-string@5.2.0",
+     "preact-render-to-string": "https://esm.sh/*preact-render-to-string@6.1.0",

      "twind": "https://esm.sh/twind@0.16.17",
      "twind/": "https://esm.sh/twind@0.16.17/"
    }
  }
```

## Automatic update checks

Fresh will periodically check if a new Fresh version is available if it's

> running outside of a CI. This happens once per day and can be disabled by

Reviewer: ```suggestion
running outside of CI. This happens once per day and can be disabled by
```
Nit
